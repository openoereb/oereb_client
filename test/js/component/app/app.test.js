import {mount} from "enzyme";
import toJson from "enzyme-to-json";
import {register} from 'ol/proj/proj4';
import proj4 from 'proj4';
import React from "react";
import {act} from "react-dom/test-utils";
import {Provider} from "react-redux";

import App from "../../../../oereb_client/static/src/component/app/app";
import MainStore from "../../../../oereb_client/static/src/store/main";

const config = {
  application: {
    title: [{
      Language: 'en',
      Text: 'PLR Cadastre, Canton of Basel-Landschaft'
    }, {
      Language: 'de',
      Text: 'ÖREB-Kataster, Kanton Basel-Landschaft'
    }, {
      Language: 'fr',
      Text: 'Cadastre RDPPF, Canton de Bâle-Campagne'
    }],
    logo_canton: [{
      Language: 'en',
      URL: 'http://example.com/logo_bl.png'
    }, {
      Language: 'de',
      URL: 'http://example.com/logo_bl.png'
    }, {
      Language: 'fr',
      URL: 'http://example.com/logo_bl.png'
    }],
    logo_oereb: [{
      Language: 'en',
      URL: 'http://example.com/logo_oereb_en.png'
    }, {
      Language: 'de',
      URL: 'http://example.com/logo_oereb_de.png'
    }, {
      Language: 'fr',
      URL: 'http://example.com/logo_oereb_fr.png'
    }],
    local_storage_prefix: 'bl',
    languages: ['en', 'de', 'fr'],
    default_language: 'en'
  },
  view: {
    map_x: 2615000,
    map_y: 1255000,
    map_zoom: 2,
    resolutions: [250, 100, 50, 20, 10, 5, 2.5, 2, 1.5, 1, 0.5, 0.25, 0.1, 0.05, 0.025, 0.01]
  },
  base_layer: {
    type: 'wms',
    url: 'http://example.com/wms',
    params: {
      LAYERS: 'base_layer',
      FORMAT: 'image/png'
    }
  },
  availability: {
    url: 'http://example.com/wms',
    params: {
      LAYERS: 'availability',
      FORMAT: 'image/png'
    }
  },
  mask_surrounding: {
    url: 'http://example.com/wms',
    params: {
      LAYERS: 'outside_bl_area',
      FORMAT: 'image/png'
    },
    opacity: 0.8
  },
  search: 'http://example.com/search',
  use_tile_wms: false
};

describe('app component', () => {

  let component;

  beforeEach(() => {
    proj4.defs(
      'EPSG:2056',
      '+proj=somerc +lat_0=46.95240555555556 +lon_0=7.439583333333333 +k_0=1 +x_0=2600000 +y_0=1200000 ' +
      '+ellps=bessel +towgs84=674.374,15.056,405.346,0,0,0,0 +units=m +no_defs'
    );
    register(proj4);
    act(() => {
      component = mount(
        <Provider store={MainStore}>
          <App config={config} />
        </Provider>
      );
    });
  });

  it('should render app element', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

});
