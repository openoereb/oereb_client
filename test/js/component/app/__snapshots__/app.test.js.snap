// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component should render app element 1`] = `
<div
  className="oereb-client-app"
>
  <div
    className="oereb-client-map-wrapper"
  >
    <div
      className="oereb-client-overlay"
    />
    <div
      className="oereb-client-map"
    />
  </div>
  <div
    className="oereb-client-information-panel container-fluid d-flex flex-column"
  >
    <div
      className="mb-4 oereb-client-information-panel-nav-mobile"
    >
      <div
        className="d-flex"
      >
        <ul
          className="nav nav-pills nav-justified flex-column"
        >
          <li
            className="nav-item"
          >
            <button
              className="nav-link active"
              onClick={[Function]}
            >
              <span
                className="text-body"
              >
                extract.information_panel.general_information.title
              </span>
            </button>
          </li>
          <li
            className="nav-item"
          >
            <button
              className="nav-link"
              onClick={[Function]}
            >
              <span
                className="text-body"
              >
                extract.information_panel.exclusion_of_liability.title
              </span>
            </button>
          </li>
          <li
            className="nav-item"
          >
            <button
              className="nav-link"
              onClick={[Function]}
            >
              <span
                className="text-body"
              >
                extract.information_panel.glossary.title
              </span>
            </button>
          </li>
        </ul>
        <div
          className="flex-grow-1 text-end ms-4"
        >
          <button
            className="btn btn-outline-secondary text-body"
            onClick={[Function]}
          >
            <i
              className="bi bi-x-lg"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="flex-grow-1 content-wrapper"
    />
  </div>
  <div
    className="oereb-client-extract-wrapper"
  />
  <div
    className="oereb-client-menu d-flex flex-column pb-3"
  >
    <div
      className="flex-grow-1 row justify-content-evenly"
    >
      <div
        className="col"
      >
        <img
          className="img-fluid p-2"
          src="http://example.com/logo_oereb_en.png"
        />
      </div>
      <div
        className="col text-end"
      >
        <img
          className="img-fluid p-2"
          src="http://example.com/logo_bl.png"
        />
      </div>
    </div>
    <div
      className="container-fluid"
    >
      <div
        className="input-group"
      >
        <div
          className="oereb-client-language-selector"
        >
          <button
            aria-expanded="false"
            className="btn btn-outline-secondary dropdown-toggle"
            data-bs-toggle="dropdown"
            title="menu.language.title"
            type="button"
          >
            EN
          </button>
          <ul
            className="dropdown-menu"
          >
            <li>
              <button
                className="dropdown-item"
                onClick={[Function]}
              >
                EN
              </button>
            </li>
            <li>
              <button
                className="dropdown-item"
                onClick={[Function]}
              >
                DE
              </button>
            </li>
            <li>
              <button
                className="dropdown-item"
                onClick={[Function]}
              >
                FR
              </button>
            </li>
          </ul>
        </div>
        <button
          aria-expanded="false"
          className="btn btn-outline-secondary dropdown-toggle"
          data-bs-toggle="dropdown"
          title="menu.settings.title"
          type="button"
        >
          <i
            className="bi bi-gear-fill"
          />
        </button>
        <ul
          className="dropdown-menu"
        >
          <li>
            <button
              className="dropdown-item"
              onClick={[Function]}
            >
              <span>
                <i
                  className="bi bi-check-square"
                />
                 
                menu.settings.availability
              </span>
            </button>
          </li>
          <li>
            <button
              className="dropdown-item"
              onClick={[Function]}
            >
              <span>
                <i
                  className="bi bi-check-square"
                />
                 
                menu.settings.symbol_zoom
              </span>
            </button>
          </li>
        </ul>
        <button
          aria-expanded="false"
          className="btn btn-outline-secondary dropdown-toggle disabled"
          data-bs-toggle="dropdown"
          title="menu.history.title"
          type="button"
        >
          <i
            className="bi bi-clock-history"
          />
        </button>
        <ul
          className="dropdown-menu"
        />
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="menu.search.placeholder"
          type="text"
          value=""
        />
        <button
          className="btn btn-outline-secondary"
          disabled={true}
          title="menu.search.reset"
          type="button"
        >
          <i
            className="bi bi-trash"
          />
        </button>
      </div>
    </div>
    <div
      className="container-fluid"
    />
  </div>
</div>
`;
